Run honeytail without just or access to docker folder

rebkwok opened the pull request:
Study repos using the research template won't have access to the docker folder, so we need to run the image directly, not via just

https://github.com/opensafely/research-template/runs/6309659194?check_suite_focus=true

rebkwok commented:
> Right, this makes sense but also didn't occur to me at all when looking at this before. I think my mental model of what's happening with Github Actions is a bit fuzzy too!
> 
> I was just checking that this docker invocation respected the `dummy-key` thing (which it does) but then I tried running it with a different key and got an error:
> 
> ```
> expected argument for flag `-d, --dataset', but got option `--backfill'
> Error: failed to parse the command line.
> 	expected argument for flag `-d, --dataset', but got option `--backfill'
> 
> Usage: honeytail -p <parser> -k <writekey> -f </path/to/logfile> -d <mydata> [optional arguments]
> 
> For even more detail on required and optional parameters, run
> honeytail --help
> ```
> 
> From a quick look at the Dockerfile I think this is because the `$DATASET` env var isn't set anywhere.

It's set by the docker-compose when it builds, and defaults to `research-action`; did you build it with a `docker build`, or are you running the already-built image?  I've definitely run it to send logs up to honeytail

evansd commented:
I ran the image as pulled directly from ghcr.io, which I _think_ is what the action will do (with the previous caveat that my mental model is a bit fuzzy here).

rebkwok commented:
> I ran the image as pulled directly from ghcr.io, which I _think_ is what the action will do (with the previous caveat that my mental model is a bit fuzzy here).

Since I changed this to not use `docker-compose run` I think I must have only tested the dev version with real pushing, to honeytail (which uses docker-compose, so gets the DATASET env var from there), and the prod version with dummy keys.  Anyway, I've updated it to set the env var, and pushed a new image.
